Error when configure --enable-linux-builtin

I've been compiling master branch for a long time now.  Somewhere, probably between 57f192fcaa4c3c90b1eea8c3bab546a942ca5de5 and 7be9fa259ed711c3b5adde783b85c92caa54007e, my default configure started throwing an error:

```
checking whether lookup_bdev() wants dev_t arg... no
checking whether lookup_bdev() wants 1 arg... configure: error: 
	*** None of the expected "lookup_bdev()" interfaces were detected.
	*** This may be because your kernel version is newer than what is
	*** supported, or you are using a patched custom kernel with
	*** incompatible modifications.
	***
	*** ZFS Version: zfs-2.3.99-275_g7be9fa259
	*** Compatible Kernels: 4.18 - 6.14
```

I could test configure options, until detecting that `--enable-linux-builtin` is the reason.

### System information
Type | Version/Name
 --- | ---
Distribution Name	| Fedora
Distribution Version	| 41 (with all dnf updates)
Kernel Version	| 6.13.8-200.fc41.x86_64
Architecture	| x86_64
OpenZFS Version	| master (7be9fa259ed711c3b5adde783b85c92caa54007e)

### Describe the problem you're observing
Cannot go past configure

### Describe how to reproduce the problem
`./configure  --enable-linux-builtin`

### Include any warning/errors/backtraces from the system logs
See above.

@dioni21 this will get you past `./configure`
```diff
diff --cc META
index 6249d5ec2,1836b806c..000000000
--- a/META
+++ b/META
diff --git a/config/kernel-blkdev.m4 b/config/kernel-blkdev.m4
index 83190c6fb..9636f44c0 100644
--- a/config/kernel-blkdev.m4
+++ b/config/kernel-blkdev.m4
@@ -421,8 +421,7 @@ AC_DEFUN([ZFS_AC_KERNEL_SRC_BLKDEV_LOOKUP_BDEV], [
 
 AC_DEFUN([ZFS_AC_KERNEL_BLKDEV_LOOKUP_BDEV], [
        AC_MSG_CHECKING([whether lookup_bdev() wants dev_t arg])
-       ZFS_LINUX_TEST_RESULT_SYMBOL([lookup_bdev_devt],
-           [lookup_bdev], [fs/block_dev.c], [
+       ZFS_LINUX_TEST_RESULT([lookup_bdev_devt], [
                AC_MSG_RESULT(yes)
                AC_DEFINE(HAVE_DEVT_LOOKUP_BDEV, 1,
                    [lookup_bdev() wants dev_t arg])
@@ -430,8 +429,7 @@ AC_DEFUN([ZFS_AC_KERNEL_BLKDEV_LOOKUP_BDEV], [
                AC_MSG_RESULT(no)
 
                AC_MSG_CHECKING([whether lookup_bdev() wants 1 arg])
-               ZFS_LINUX_TEST_RESULT_SYMBOL([lookup_bdev_1arg],
-                   [lookup_bdev], [fs/block_dev.c], [
+               ZFS_LINUX_TEST_RESULT([lookup_bdev_1arg], [
                        AC_MSG_RESULT(yes)
                        AC_DEFINE(HAVE_1ARG_LOOKUP_BDEV, 1,
                            [lookup_bdev() wants 1 arg])
diff --git a/config/kernel-insert-inode-locked.m4 b/config/kernel-insert-inode-locked.m4
index 348aff9a5..b224c4074 100644
--- a/config/kernel-insert-inode-locked.m4
+++ b/config/kernel-insert-inode-locked.m4
@@ -12,8 +12,7 @@ AC_DEFUN([ZFS_AC_KERNEL_SRC_INSERT_INODE_LOCKED], [
 
 AC_DEFUN([ZFS_AC_KERNEL_INSERT_INODE_LOCKED], [
        AC_MSG_CHECKING([whether insert_inode_locked() is available])
-       ZFS_LINUX_TEST_RESULT_SYMBOL([insert_inode_locked],
-           [insert_inode_locked], [fs/inode.c], [
+       ZFS_LINUX_TEST_RESULT([insert_inode_locked], [
                AC_MSG_RESULT(yes)
        ], [
                ZFS_LINUX_TEST_ERROR([insert_inode_locked()])
diff --git a/config/kernel-truncate-setsize.m4 b/config/kernel-truncate-setsize.m4
index 76c82ef30..22ae75d2d 100644
--- a/config/kernel-truncate-setsize.m4
+++ b/config/kernel-truncate-setsize.m4
@@ -12,8 +12,7 @@ AC_DEFUN([ZFS_AC_KERNEL_SRC_TRUNCATE_SETSIZE], [
 
 AC_DEFUN([ZFS_AC_KERNEL_TRUNCATE_SETSIZE], [
        AC_MSG_CHECKING([whether truncate_setsize() is available])
-       ZFS_LINUX_TEST_RESULT_SYMBOL([truncate_setsize],
-           [truncate_setsize], [mm/truncate.c], [
+       ZFS_LINUX_TEST_RESULT([truncate_setsize], [
                AC_MSG_RESULT(yes)
        ], [
                ZFS_LINUX_TEST_ERROR([truncate_setsize])
```
However when I actually go into the kernel source and build with ZFS included, I'm seeing further errors:
```
  CC      fs/zfs/zfs/ddt.o
  CC      fs/zfs/zfs/ddt_log.o
  CC      fs/zfs/zfs/ddt_stats.o
In file included from ./include/zfs/os/linux/zfs/sys/trace_acl.h:36,
                 from ./include/zfs/os/linux/zfs/sys/trace_zfs.h:40,
                 from fs/zfs/zfs/arc.c:307:
./include/zfs/os/linux/kernel/linux/vfs_compat.h:170:1: error: conflicting types for ‘setattr_prepare’; have ‘int(struct dentry *, struct iattr *)’
  170 | setattr_prepare(struct dentry *dentry, struct iattr *ia)
      | ^~~~~~~~~~~~~~~
In file included from ./include/zfs/os/linux/spl/sys/vfs.h:29,
                 from ./include/zfs/os/linux/spl/sys/cred.h:32,
                 from ./include/zfs/os/linux/spl/sys/zone.h:29,
                 from ./include/zfs/os/linux/spl/sys/sysmacros.h:33,
                 from ./include/zfs/sys/zfs_context.h:47,
                 from ./include/zfs/sys/spa.h:39,
                 from fs/zfs/zfs/arc.c:287:
```
I'm still looking into it

--- a/config/kernel.m4
+++ b/config/kernel.m4
@@ -899,7 +899,7 @@ AC_DEFUN([ZFS_LINUX_TEST_RESULT_SYMBOL], [
 					eval "$cachevar=no"
 				])
 			], [
-				eval "$cacheval=yes"
+				eval "$cachevar=yes"
 			])
 		])
 	])

